frotz/src/interface/frotz_interface.c

```diff
@@ -1704,18 +1704,17 @@ void test() {
 // Given a list of action candidates, find the ones that lead to valid world changes.
 // candidate_actions contains a string with all the candidate actions, seperated by ';'
 // valid_actions will be written with each of the identified valid actions seperated by ';'
-// equiv_classes will be written with the world_diff index for each valid_action indicating
+// diff_array will be written with the world_diff for each valid_action indicating
 // which of the valid actions are equivalent to each other in terms of their world diffs.
 // Returns the number of valid actions found.
-int filter_candidate_actions(char *candidate_actions, char *valid_actions, unsigned char *equiv_classes) {
```

Comment:
The main difference with your C function is it keeps the world_diff for each valid action. I merge them after in the Python code. The main reason being when I do the parallelization, I split the candidate actions in `os.cpucount()` chunks which are passed to the C function. That means I can't do the full equiv_classes check as you did.